EventProcessor/EventProcessor.WorkerRole/Program.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Autofac;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.EventProcessor.WorkerRole.Processors;
+using Microsoft.Azure.IoT.Samples.EventProcessor.WorkerRole.Processors;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.EventProcessor.WorkerRole
+{
+    public static class Program
+    {
+        static CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+        static IContainer eventProcessorContainer;
+
+        private const string SHUTDOWN_FILE_ENV_VAR = "WEBJOBS_SHUTDOWN_FILE";
+        private static bool _shutdownSignalReceived = false;
+        private static string _shutdownFile;
+
+        static void Main(string[] args)
+        {
+            try
+            {
+                //Cloud deploys often get staged and started to warm them up, then get a shutdown
+                //signal from the framework before being moved to the production slot. We don't want 
+                //to start initializing data if we have already gotten the shutdown message, so we'll 
+                //monitor it. This environment variable is reliable
+                //http://blog.amitapple.com/post/2014/05/webjobs-graceful-shutdown/#.VhVYO6L8-B4
+                _shutdownFile = Environment.GetEnvironmentVariable(SHUTDOWN_FILE_ENV_VAR);
+
```

Comment:
I'm not sure we need the shutdown file logic in this webjob--it was intended to prevent multiple data initializations running in parallel, but we don't have that code in this webjob.